core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/DistinctAggregator.java

```diff
@@ -106,6 +108,17 @@ public void prepareRun(BatchAggregatorContext context) throws Exception {
     validate(context.getInputSchema(), conf.getFields(), context.getFailureCollector());
     context.getFailureCollector().getOrThrowException();
 
+    if (conf.getFields() == null) {
+      context.record(TransformLineageRecorderUtils.generateOneToOnes(
+        TransformLineageRecorderUtils.getFields(context.getInputSchema()),
```

Comment:
You can just create this field with `List<String> fields = conf.getFields() == null ?  ... : ...` to avoid the if else block